ocs_samples/basic_samples/OmfIngress/DotNet/readme.md

```diff
@@ -0,0 +1,173 @@
+	}
+```
+
+The authentication values are provided to the ``OSIsoft.Identity.AuthenticationHandler``. 
+The AuthenticationHandler is a DelegatingHandler that is attached to an HttpClient pipeline.
+
+Other Configuration
+-------------------
+
+The aforementioned appsettings.json file has placeholders for the names of the connection, as well as a client Id to map a device to the topic. You must fill in those values as well.
+
+Set up IngressService
+----------------------
+
+The example works through one interface: 
+
+* IOmfIngress for for configuring OMF Connections and sending OMF Messages
+
+The following code block illustrates how to configure the IngressService to use throughout the sample:
+
+```
+    AuthenticationHandler authenticationHandler = new AuthenticationHandler(address, clientId, clientSecret);
+
+    OmfIngressService baseOmfIngressService = new OmfIngressService(new Uri(address), null, HttpCompressionMethod.None, authenticationHandler);
+    IOmfIngressService omfIngressService = baseOmfIngressService.GetOmfIngressService(tenantId, namespaceId);
+```
+
+Note that the instance of the IOmfIngressService is scoped to a tenant and namespace. If you wish to work in a different tenant or namespace, you would need another instance scoped to that tenant and namespace.
+
+OMF Connections
+-------------------------
+An OMF Connection is made up of three components: one or more Clients, a Topic, and a Subscription. Data is sent to a Topic via a Client, where the data is buffered and made available for the Subscription. The 
+Subscription relays data from the Topic to the Sequential Data Store in the namespace that the Subscription resides in. The OmfIngressClient in this example creates a connection using an existing clientId, and 
+creating a Topic and Subscription as detailed below.
+
+Clients
+---------------
+
+Devices sending OMF messages each need their own unique clientId and clientSecret. The clientId and secret are used to authenticate the requests, and the clientId is used route messages to the proper topic(s). 
+ClientIds may be mapped to at most one topic per namespace. For more details on Clients see the [Client Credential Client Documentation](https://ocs-docs.osisoft.com/Documentation/Identity/Identity_ClientCredentialClient.html).
+
+Topics
+--------------
+
+A Topic is used to aggregate data received from clients and make it available for consumption 
+via a Subscription. A topic must contain at least one client Id. Client Ids may be added to 
+or removed from an existing topic. First, we create the Topic locally by instantiating 
+a new Topic object:
+
+
+```
+    Topic topic = new Topic()
+    {
+        Name = "REPLACE_WITH_TOPIC_NAME",
+        Description = "This is a sample Topic",
+        ClientIds = new List<string>() { mappedClientId }
+    };
+```
+
+Then use the Ingress client to create the Topic in OCS:
+
+```
+    Topic createdTopic = await ingressService.CreateOrUpdateTopicAsync(topic);
```

Comment:
This should also be omfIngressService. I would check all the files to make sure we use omfIngressService and omfIngressClient.